### PR TITLE
feat: support ESLint 8

### PR DESCRIPTION
**Pull request by DiegoRBaquero**
This PR updates eslint-plugin-jsdoc and eslint-plugin-promise specifically, which are the only blockers for ESLint 8 adoption. Also fixes repository url.

These changes seems to be non-breaking, i.e. still fully supported on ESLint 7

### CODE REVIEW

**Comment by achingbrain** [comment]
It seems there are some incompatible peer dependencies introduced by this PR - @DiegoRBaquero could you take a look please?

**Comment by DiegoRBaquero** [comment]
Taking a look, blocked by https://github.com/standard/eslint-config-standard-with-typescript/issues/792

**Comment by BigLep** [comment]
2022-05-27: we'll check back in on this in a week or two.  If https://github.com/standard/eslint-config-standard-with-typescript/issues/792 isn't resolved, we'll ask the author to unblock themselves by publishing a fork.

**Comment by BigLep** [comment]
2022-06-10: we will continue to let this sit given the active work in https://github.com/standard/eslint-config-standard-with-typescript/issues/792#issuecomment-1151962821

**Comment by BigLep** [comment]
@DiegoRBaquero : given https://github.com/standard/eslint-config-standard-with-typescript/issues/792 has been resolved, do we need to do any updates or just trigger the build again?

**Comment by achingbrain** [comment]
Yes, the `eslint-config-standard-with-typescript` dep needs updating to `^22.0.0` and the `package-lock.json` file needs regenerating.

File: test/validate.js [hunk 1]
```diff
@@ -2,14 +2,17 @@ const eslint = require("eslint")
 const test = require("tape")
 const eslintrc = require("..")
 
-test("load config in eslint to validate all rule syntax is correct", function (t) {
-  var cli = new eslint.CLIEngine({
+console.info(eslint)
+
+test("load config in eslint to validate all rule syntax is correct", async function (t) {
+  var cli = new eslint.ESLint({
```
Comment (achingbrain) [inline test/validate.js:8]:
```suggestion
test("load config in eslint to validate all rule syntax is correct", async function (t) {
  var cli = new eslint.ESLint({
```

**Pull request merged**